src/sqlfluff/core/rules/functional/segments.py

```diff
@@ -0,0 +1,118 @@
+"""Surrogate class for working with Segment collections."""
+from typing import Any, Callable, List, Optional
+
+from sqlfluff.core.parser import BaseSegment
+from sqlfluff.core.templaters.base import TemplatedFile
+from sqlfluff.core.rules.functional.raw_file_slices import RawFileSlices
+
+
+class Segments(list):
+    """Encapsulates a sequence of one or more BaseSegments.
+
+    The segments may or may not be contiguous in a parse tree.
+    Provides useful operations on a sequence of segments to simplify rule creation.
+    """
+
+    def __new__(cls, templated_file, *segments):
+        """Override new operator."""
+        return super(Segments, cls).__new__(cls, segments)
+
+    def __init__(self, templated_file: Optional[TemplatedFile], *segments: BaseSegment):
+        super().__init__(segments)
+        self.templated_file = templated_file
+
+    def __add__(self, segments) -> "Segments":
+        return Segments(self.templated_file, *super().__add__(segments))
+
+    def __radd__(self, segments) -> "Segments":
+        return Segments(self.templated_file, *list(segments).__add__(list(self)))
+
+    def all(self, predicate: Optional[Callable[[BaseSegment], bool]] = None) -> bool:
+        """Do all the segments match?"""
+        for s in self:
+            if predicate and not predicate(s):
```

Comment:
```suggestion
            if predicate is not None and not predicate(s):
```
small nitpick here: when comparing to None as a default argument you should use the explicit `predicate is not None` (stops any weirdness if the user puts `predicate = 0` or `predicate = []` etc as they are falsey also).

There's a couple of other predicates that need this changing also

Reply:
Will do!